backend/user-service/src/controllers/auth.controller.ts

```diff
@@ -67,3 +75,91 @@ export async function hashPassword(password: string): Promise<string> {
     const saltRounds = 10
     return hash(password, saltRounds)
 }
+
+export async function getHTMLTemplate(htmlFilePath: string): Promise<string> {
+    const filePath = path.join(__dirname, htmlFilePath)
+    return await fs.readFile(filePath, 'utf8')
+}
+
+export async function sendMail(to: string, subject: string, text: string, html: string): Promise<void> {
+    const transporter = nodemailer.createTransport({
+        service: 'gmail',
+        host: 'smtp.gmail.com',
+        port: 587,
+        secure: false,
+        auth: {
+            user: process.env.EMAIL,
+            pass: process.env.PASSWORD,
+        },
+    })
+    await transporter.sendMail({
+        from: process.env.EMAIL,
+        to,
+        subject,
+        text,
+        html,
+    })
+}
+
+export function generateOTP(): string {
+    return Math.floor(100000 + Math.random() * 900000).toString()
+}
+
+export async function handleReset(request: TypedRequest<EmailVerificationDto>, response: Response): Promise<void> {
+    const createDto = EmailVerificationDto.fromRequest(request)
+    createDto.verificationToken = ''
+    const errors = await createDto.validate()
+    if (errors.length) {
+        const errorMessages = errors.map((error: ValidationError) => `INVALID_${error.property.toUpperCase()}`)
+        response.status(400).json(errorMessages).send()
+        return
+    }
+
+    const user = await findOneUserByEmail(createDto.email)
+
+    if (!user) {
+        response.status(404).json('USER_NOT_FOUND').send()
+        return
+    }
+    if (user.verificationToken !== '') {
+        response.status(400).json('TOKEN_ALREADY_SENT').send()
+        return
+    }
+
+    const otp = generateOTP()
+    createDto.verificationToken = otp
+
+    await updateUser(user.id, createDto)
+
+    const htmlFile = await getHTMLTemplate('../util/passwordResetEmail.html')
```

Comment:
Have you tested this? When the TS code gets transpiled does the html file also go into the dist directory?

Reply:
good point. just handled it. now i have placed the html file in the public directory (outside the src directory), so it will work even after the ts gets transpiled.